pkg/kubelet/cm/dra/manager.go

```diff
@@ -226,14 +226,17 @@ func (m *ManagerImpl) prepareResources(ctx context.Context, pod *v1.Pod) error {
 			// Get a reference to the claim info for this claim from the cache.
 			// If there isn't one yet, then add it to the cache.
 			claimInfo, exists := m.cache.get(resourceClaim.Name, resourceClaim.Namespace)
-			if !exists {
+			switch {
+			case !exists:
 				ci, err := newClaimInfoFromClaim(resourceClaim)
 				if err != nil {
 					return fmt.Errorf("claim %s: %w", klog.KObj(resourceClaim), err)
 				}
 				claimInfo = m.cache.add(ci)
 				logger.V(6).Info("Created new claim info cache entry", "pod", klog.KObj(pod), "podClaim", podClaim.Name, "claim", klog.KObj(resourceClaim), "claimInfoEntry", claimInfo)
-			} else {
+			case claimInfo.ClaimUID != resourceClaim.UID:
+				return fmt.Errorf("old claim with same name %s and different UID %s still exists (previous pod force-deleted?!)", klog.KObj(resourceClaim), claimInfo.ClaimUID)
+			default:
 				logger.V(6).Info("Found existing claim info cache entry", "pod", klog.KObj(pod), "podClaim", podClaim.Name, "claim", klog.KObj(resourceClaim), "claimInfoEntry", claimInfo)
 			}
```

Comment:
I think we should capture the last log even if claimInfo.ClaimUID != resourceClaim.UID.

As such, I think you can leave the existing code as is, and just add the following:
```
if claimInfo.ClaimUID != resourceClaim.UID {
	return fmt.Errorf("old claim with same name %s and different UID %s still exists (previous pod force-deleted?!)"
}
```
In the case where the claim didn't previously exist in the cache, we will have just inserted it, so we know this check can never be true.

Reply:
That was my initial implementation that triggered [this discussion](https://github.com/kubernetes/kubernetes/pull/131617#discussion_r2074876967).

Let me know if you still want me to do that.

@pohly

Reply:
I actually don't feel too strongly either way. If @pohly feels strongly about a switch statement then that is fine with me (I had missed his comments since they weren't in the diff anymore when I looked after you had updated).

Reply:
So, what to do? I don't feel strongly about this, just want it to be merged.

Reply:
I defer to @klueska. He needs to approve anyway.